emcc.py

```diff
@@ -2074,19 +2088,26 @@ def un_src(): # use this if you want to modify the script and need it to be inli
           xhr.send(null);
 ''' % (os.path.basename(wasm_binary_target), script_inline)
 
-      html = open(target, 'w')
+      html = open(target, 'wb')
       assert (script_src or script_inline) and not (script_src and script_inline)
       if script_src:
         script_replacement = '<script async type="text/javascript" src="%s"></script>' % script_src
       else:
         script_replacement = '<script>\n%s\n</script>' % script_inline
-      html.write(shell.replace('{{{ SCRIPT }}}', script_replacement))
+      html_contents = shell.replace('{{{ SCRIPT }}}', script_replacement)
+      html_contents = tools.line_endings.convert_line_endings(html_contents, '\n', output_eol)
+      html.write(html_contents)
       html.close()
     else: # final_suffix != html
       if proxy_to_worker:
         shutil.move(js_target, js_target[:-3] + '.worker.js') # compiler output goes in .worker.js file
         worker_target_basename = target_basename + '.worker'
-        open(target, 'w').write(open(shared.path_from_root('src', 'webGLClient.js')).read() + '\n' + open(shared.path_from_root('src', 'proxyClient.js')).read().replace('{{{ filename }}}', shared.Settings.PROXY_TO_WORKER_FILENAME or worker_target_basename).replace('{{{ IDBStore.js }}}', open(shared.path_from_root('src', 'IDBStore.js')).read()))
+        target_contents = open(shared.path_from_root('src', 'webGLClient.js')).read() + '\n' + open(shared.path_from_root('src', 'proxyClient.js')).read().replace('{{{ filename }}}', shared.Settings.PROXY_TO_WORKER_FILENAME or worker_target_basename).replace('{{{ IDBStore.js }}}', open(shared.path_from_root('src', 'IDBStore.js')).read())
+        target_contents = tools.line_endings.convert_line_endings(convert_line_endings, '\n', output_eol)
+        open(target, 'wb').write(target_contents)
+
+    if os.path.isfile(asm_target): tools.line_endings.convert_line_endings_in_file(asm_target, os.linesep, output_eol)
```

Comment:
this looks risky. perhaps `asm_target` wasn't used but already exists on disk? ditto for `js_target`. it seems like we should know when we used those files (e.g. set a var to state that), and do this only then.